packages/moleculer-db/src/index.js

```diff
@@ -886,14 +887,22 @@ module.exports = {
 
 			if (this.settings.useDotNotation)
 				sets = flatten(sets, { safe: true });
+			
+			if(!id && params.where && params.update) {
+				return this.adapter.updateMany(params.where, params.update);
+			}
 
-			return this.adapter.updateById(id, { "$set": sets })
-				.then(doc => {
-					if (!doc)
-						return Promise.reject(new EntityNotFoundError(id));
-					return this.transformDocuments(ctx, {}, doc)
-						.then(json => this.entityChanged("updated", json, ctx).then(() => json));
-				});
+			if(id) {
+				return this.adapter.updateById(id, { "$set": sets })
+					.then(doc => {
+						if (!doc)
+							return Promise.reject(new EntityNotFoundError(id));
+						return this.transformDocuments(ctx, {}, doc)
+							.then(json => this.entityChanged("updated", json, ctx).then(() => json));
+					});
+			}
+			
+			return Promise.reject(new ValidationError("invalid_params"));
```

Comment:
Please use more accurate errors, Correct message text, type, code...etc.

Reply:
I have added a more accurate error message and params as data. I have also added the missing test for failing the _update and _remove method